### PR TITLE
[Tiny PR] RichText: correctly update value after undo

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

See #17839. RichText is not removing formatting after undo. This is caused by `updateFormats`, which seems to be called in that place for no reason. Looks like I made a mistake calling it there.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by ellatrix** [comment]
Seems to be an issue with RichText not applying the change.

**Comment by ellatrix** [comment]
Seems to be introduced in #15496.

File: packages/rich-text/src/component/index.js [hunk 1]
```diff
@@ -843,21 +843,11 @@ class RichText extends Component {
...
 
-		const { activeFormats = [] } = this.record;
-
 		if ( shouldReapply ) {
 			this.value = value;
 			this.record = this.formatToValue( value );
 			this.record.start = selectionStart;
 			this.record.end = selectionEnd;
-
-			updateFormats( {
```
Comment (mcsf) [inline packages/rich-text/src/component/index.js]: So was `updateFormats` mutating the record back to a state with the previous formats? Is that it?

**Review by mcsf — APPROVED** [review]
I haven't tested this extensively, but it does the job for me.

File: packages/rich-text/src/component/index.js (hunk 1 above)
Comment (ellatrix) [reply packages/rich-text/src/component/index.js]: Yes, that's it. :) I'm not sure why it's here, I probably made a mistake.

**Comment by ellatrix** [comment]
Thanks for the review!

**Pull request merged**